include/seqan3/core/detail/debug_stream_range.hpp

```diff
@@ -17,6 +17,52 @@
 #include <seqan3/core/type_traits/range.hpp>
 #include <seqan3/std/ranges>
 
+namespace seqan3::detail
+{
+/*!\brief A helper concept definition for ranges that can be streamed to the seqan3::debug_stream.
+ * \tparam rng_t The range type to check.
+ * \ingroup core
+ *
+ * \details
+ *
+ * This concept refines the std::ranges::InputRange concept to allow streaming the range object to the debug stream,
+ * with the following requirements:
+ *
+ * * `rng_t` is not the same type as `reference_t<rng_t>`,
+ * * `rng_t` is not a pointer or c-style array,
+ * * `reference_t<rng_t>` is not `char`.
+ *
+ * \note If you need to further refine this concept for your type ensure that this concept comes before any
+ *       additional refinement, otherwise your concept definition/requires clause is ill-formed according to the
+ *       subsumption rules for concepts.
+ */
+template <typename rng_t>
+SEQAN3_CONCEPT DebugStreamableRange =
+    !std::Same<remove_cvref_t<reference_t<rng_t>>, remove_cvref_t<rng_t>> && // prevent recursive instantiation
+    // exclude null-terminated strings:
+    !(std::is_pointer_v<std::decay_t<rng_t>> && std::Same<remove_cvref_t<reference_t<rng_t>>, char>);
```

Comment:
You removed `requires (reference_t<rng_t> l) { { s << l }; }` right? I imagine that that caused problems.

Because there is no debug_stream included in DebugStreamableRange and thus makes the name confusing.

Reply:
that was something you want to statically assert for but not make overload resolution. if the type is not streamable to the stream, what would be the correct overload then? So I don't see the point of making this a constraint.

>Because there is no debug_stream included in DebugStreamableRange and thus makes the name confusing.

I don't understand what you mean?!

Reply:
The name is DebugStreamableRange but there are no checks for debug_streamable

Reply:
Ok, the name is weird. I can change this. I didn't want to expose these restrictions into the public user interface but they are necessary since you want to call another overload if it does not fulfill the range requirements for our debug_stream overload (e.g. filesystems directory_iterator). But that might be a problem as well if we allow users to add their own types to be printable to the debug_stream and they happen to be a range as well.